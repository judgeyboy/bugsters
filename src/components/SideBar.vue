<template>
  <div class="off-canvas-sidebar" :class="showSidebar ? 'active': ''">
    <div class="bugster-sidebar">

      <div class="bugster-sidebar__menu">
        <slot></slot>
      </div>

      <div class="bugster-sidebar__language">
        <LanguageSwitcher />
      </div>
    </div>
  </div>

</template>

<script>
import LanguageSwitcher from './LanguageSwitcher'

export default {
  name: 'SideBar',
  props: ['showSidebar'],
  components: {
    LanguageSwitcher
  }
}
</script>

<style lang="sass" scoped>
.off-canvas-sidebar
  width: 12rem
  min-height: 100vh

.bugster-sidebar
  bottom: 1.5rem
  -webkit-overflow-scrolling: touch
  overflow-y: auto
  padding: .5rem 1.5rem
  position: fixed
  top: 3.5rem
  width: 12rem

  &__language
    position: fixed
    bottom: 3.8rem
</style>
