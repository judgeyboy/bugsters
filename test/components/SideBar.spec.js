import { shallow } from 'vue-test-utils'

import SideBar from '@/components/SideBar'

describe('SideBar component', () => {
  it('should be a vue instance', () => {
    const wrapper = shallow(SideBar)
    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('should match snapshot', () => {
    const wrapper = shallow(SideBar)
    const $html = wrapper.vm.$el.outerHTML
    expect($html).toMatchSnapshot()
  })
})
