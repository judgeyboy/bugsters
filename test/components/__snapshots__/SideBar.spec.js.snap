// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar component should match snapshot 1`] = `

<div class="off-canvas-sidebar">
  <div class="bugster-sidebar">
    <div class="bugster-sidebar__menu">
    </div>
    <div class="bugster-sidebar__language">
    </div>
  </div>
</div>

`;
